godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -367,6 +368,13 @@ bool SurfaceDetection::find_surfaces()
 	std::vector<pcl::PointIndices> clusters_indices;
 	std::vector<Normals::Ptr> segment_normals;
 
+	// Pass through filter the data to constrain it to our ROI
+  pcl::PassThrough<pcl::PointXYZ> pass;
+  pass.setInputCloud(process_cloud_ptr);
+  pass.setFilterFieldName("z");
+  pass.setFilterLimits(-0.1, 0.4);
+  pass.filter(*process_cloud_ptr);
+
```

Comment:
This was added to keep the amount of data from the Kinect2 under control. The voxel downsampler was failing because the point cloud was too large.